scripts/sanity_chk/sanitylib.py

```diff
@@ -2587,6 +2589,7 @@ def save_reports(self, name, suffix, report_dir, no_update, release, only_failed
             self.xunit_report(filename + ".xml", full_report=False, append=only_failed)
             self.xunit_report(filename + "_report.xml", full_report=True, append=only_failed)
             self.csv_report(filename + ".csv")
+            self.json_report(filename + ".json")
```

Comment:
I see no handling of "only_failed" running mode. It allows rerunning only failed tests and modify the results of failed tests instead of rewriting the whole report. In the current version of PR if e.g. there were 8 test suites and one failed and one called sanitycheck with "--only-failed" the report with 8 suites will be replaced with the newest one with only one suite. self.xunit_report() deals with it by "append=only_failed".

Reply:
> I think there is a bug now in reporting @nashif. sanitycheck.xml report behaves the same as this json report. The summary entry states the correct number of tests executed but inside there are only entries from retried tests. Other reports (sanitycheck_report.xml and platform.xml) are correct, they have all the entries

file a bug please.